python/dgl/graphbolt/item_sampler.py

```diff
@@ -216,7 +220,7 @@ def __iter__(self):
             buffer = self._item_set[start_offset + start : start_offset + end]
             indices = torch.arange(end - start)
             if self._shuffle:
-                np.random.shuffle(indices.numpy())
+                self._rng.shuffle(indices.numpy())
```

Comment:
```suggestion
                seed = self._rng.integers(low=0)
                torch_rng = torch.random.manual_seed(seed)
                permutation = torch.randperm(len(indices), generator=torch_rng)
                indices = indices[permutation]
```

Reply:
I thought of something like this.

Reply:
actually, since we have contiguous range, it can be improved, let me edit the suggestion above.

Reply:
Later, we can add a device argument to this class and it can be made to work on the GPU easily, if we decide to do so.

Reply:
what's the benefit of using `torch` and `np` together?

Reply:
We can store np as a member of the class since it can be pickled. We use torch for the actual operation so that it can be made to work on the GPU later if we decide to do so.

Reply:
If we really need to turn to `torch.Generator()` some day, we could refactor the `ItemSampler`.

Reply:
We can pass device arguments to torch operations.

Reply:
With my suggestion above, we don't need to turn to torch.Generator. We can keep the np generator as before. And this part wouldn't even need to change if we refactor this code as it will already be working via torch operations.

Reply:
Is there any disadvantage going forward with my suggestion above? I feel that as a torch library, we should use torch for almost all operations if we can so that everything can utilize torch internals, torch memory pools etc. even if things run on the CPU.

Reply:
Looks like numpy's way of shuffling is more performance on the CPU, so maybe we can do this in a later PR if we decide to run the operation on the GPU.